pinot-common/src/main/java/com/linkedin/pinot/common/config/SegmentPartitionConfig.java

```diff
@@ -51,9 +52,32 @@ public SegmentPartitionConfig(
    * @return Partition function for the column.
    */
   @Nullable
-  public PartitionFunction getPartitionFunction(@Nonnull String column) {
+  public String getFunctionName(@Nonnull String column) {
     ColumnPartitionConfig columnPartitionConfig = _columnPartitionMap.get(column);
-    return (columnPartitionConfig != null) ? columnPartitionConfig.getPartitionFunction() : null;
+    return (columnPartitionConfig != null) ? columnPartitionConfig.getFunctionName() : null;
+  }
+
+  /**
+   * Set the number of partitions for the specified column.
+   *
+   * @param column Column for which to set the number of partitions.
+   * @param numPartitions Number of partitions to set.
+   */
+  @JsonIgnore
+  public void setNumPartitions(String column, int numPartitions) {
```

Comment:
If code is designed to handle different number of partitions per column. There are calls for both functions.